src/Components/ClustersListTable/ClustersListTable.js

```diff
@@ -0,0 +1,247 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  SortByDirection,
+  Table,
+  TableBody,
+  TableHeader,
+  TableVariant,
+} from '@patternfly/react-table';
+import { Card, CardBody } from '@patternfly/react-core/dist/js/components/Card';
+import {
+  Pagination,
+  PaginationVariant,
+} from '@patternfly/react-core/dist/js/components/Pagination';
+import PrimaryToolbar from '@redhat-cloud-services/frontend-components/PrimaryToolbar/PrimaryToolbar';
+
+import {
+  CLUSTERS_LIST_INITIAL_STATE,
+  updateClustersListFilters,
+} from '../../Services/Filters';
+import {
+  CLUSTERS_LIST_COLUMNS,
+  CLUSTER_FILTER_CATEGORIES,
+} from '../../AppConstants';
+import {
+  buildFilterChips,
+  mapClustersToRows,
+  passFiltersCluster,
+} from '../Common/Tables';
+import Loading from '../Loading/Loading';
+import messages from '../../Messages';
+import { ErrorState, NoMatchingClusters } from '../MessageState/EmptyStates';
+
+const ClustersListTable = ({ query }) => {
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const updateFilters = (payload) =>
+    dispatch(updateClustersListFilters(payload));
```

Comment:
@karelhala, do you really think that simple one-line functions like the `dispatch` here really require a `useCallback` wrapping? Furthermore, the function `updateFilters` is not listed in any of the `useEffect` dependencies == the re-calculating of the instance of this function does not trigger any further re-renders.

But this is an interesting question, since I have recently read this article https://dmitripavlutin.com/dont-overuse-react-usecallback/. It turns out, the `useCallback` is often overused. Just FYI.

Reply:
@gkarat it depends on the complexity of the function you want to memorize. In this case, I agree, the `updateClustersListFilters` function is very simple. Adding the hook around it probably just makes the code slower.